Ensure tf.clip_by_value returns positive zero for clipped values on GPU

zvoicu000 opened the pull request:
Fixes #99759

This PR resolves an inconsistency between CPU and GPU for tf.clip_by_value when clipping -0.0. Now, both devices return positive zero (0.0) for clipped values, ensuring consistent behavior.

zvoicu000 committed:
Ensure tf.clip_by_value returns positive zero for clipped values on GPU

mihaimaruseac reviewed (changes requested):
Please add a test and also a link to a real bug where this matters

zvoicu000 committed:
Add GPU test for tf.clip_by_value positive zero; reference bugs #67279 and #99759

- Adds a test to ensure tf.clip_by_value returns +0.0 for clipped values on GPU.
- Adds comments referencing real bugs: https://github.com/tensorflow/tensorflow/issues/67279 and https://github.com/tensorflow/tensorflow/issues/99759.

mihaimaruseac commented:
If there's no use case where this is important, then it does not make sense to do it as it affects performance.
